sdk/common/src/main/java/io/opentelemetry/sdk/internal/AttributesMap.java

```diff
@@ -18,6 +18,14 @@
  * A map with a fixed capacity that drops attributes when the map gets full, and which truncates
  * string and array string attribute values to the {@link #lengthLimit}.
  *
+ * <p>WARNING: In order to reduce memory allocation, this class extends {@link HashMap} when it
+ * would be more appropriate to delegate. The problem with extending is that we don't enforce that
+ * all {@link HashMap} methods for reading / writing data conform to the configured attribute
+ * limits. Therefore, it's easy to accidentally call something like {@link Map#putAll(Map)} or
+ * {@link Map#put(Object, Object)} and bypass the restrictions (see <a
```

Comment:
is `Map#put(Object, Object)` still a problem now that you've changed the overload to an override?

Reply:
Ah no. Will update to use `putAll` as an example.